errors.go

```diff
@@ -18,3 +18,38 @@ func (e errorList) Error() string {
 	}
 	return "multiple errors:\n" + strings.Join(msgs, "\n")
 }
+
+// IgnorableError is an error that has an extra method to tell whether it should
+// be ignored. Ignored errors may be omitted from standard program output.
+type IgnorableError interface {
+	error
+	Ignore() bool
+}
+
+// ignoredError implements IgnorableError, always returning true.
+type ignoredError struct {
+	err error
+}
+
+func (e *ignoredError) Error() string {
+	if e.err == nil {
+		return ""
+	}
+	return e.err.Error()
+}
+
+func (e *ignoredError) Ignore() bool {
+	return true
+}
+
+var _ IgnorableError = (*ignoredError)(nil)
```

Comment:
To ensure in compile time that `*ignoredError` implements the `IgnorableError` interface.